lib/cpp/src/thrift/transport/TSocket.cpp

```diff
@@ -569,7 +569,7 @@ uint32_t TSocket::read(uint8_t* buf, uint32_t len) {
         throw TTransportException(TTransportException::INTERRUPTED, "Interrupted");
       }
     } else /* ret == 0 */ {
-      GlobalOutput.perror("TSocket::read() THRIFT_EAGAIN (timed out) after %f ms", recvTimeout_);
+      GlobalOutput.printf("TSocket::read() THRIFT_EAGAIN (timed out) after %d ms", recvTimeout_);
```

Comment:
Dear @stiga-huang can you kindly help me understand why `printf` is better here than `perror`?

Reply:
Sorry that I just mentioned it in the JIRA (https://issues.apache.org/jira/browse/THRIFT-5670). We saw a strange log when using cpp lib of thrift-0.16.0:
```
I1114 15:50:53.952184 16548 thrift-util.cc:196] TSocket::read() THRIFT_EAGAIN (timed out) after %f ms: Unknown error 30000
```
`recvTimeout_` (30000) is incorrectly used as `errno_copy` in `GlobalOutput.perror()`. `%f` is also not substituted as expected.
https://github.com/apache/thrift/blob/0.16.0/lib/cpp/src/thrift/TOutput.h#L40-L45
`printf` is what exactly we need. BTW, `recvTimeout_` is a `int` so replace `%f` to `%d`.

Reply:
This looks good, should be merged.